pkg/oc/generate/app/cmd/resolve.go

```diff
@@ -263,18 +277,18 @@ func DetectSource(repositories []*app.SourceRepository, d app.Detector, g *Gener
 }
 
 // AddComponentInputsToRefBuilder set up the components to be used by the reference builder.
-func AddComponentInputsToRefBuilder(b *app.ReferenceBuilder, r *Resolvers, c *ComponentInputs, g *GenerationInputs) error {
+func AddComponentInputsToRefBuilder(b *app.ReferenceBuilder, r *Resolvers, c *ComponentInputs, g *GenerationInputs, s, i *[]string) error {
 	// lookup source repositories first (before processing the component inputs)
-	repositories, err := AddSourceRepositoriesToRefBuilder(b, c.SourceRepositories, g)
+	repositories, err := AddSourceRepositoriesToRefBuilder(b, c, g, s, i)
 	if err != nil {
 		return err
 	}
-
 	// identify the types of the provided source locations
-	if err := DetectSource(repositories, r.Detector, g); err != nil {
-		return err
+	if git.IsGitInstalled() {
+		if err := DetectSource(repositories, r.Detector, g); err != nil {
```

Comment:
so no source detection is done if the source component is a local directory?

Reply:
yes, you right. forgot about this case for code discovery.
Will fix this. 
Do we have all the oc new-app behavior documented somewhere apart `-h`.

Reply:
> Do we have all the oc new-app behavior documented somewhere apart -h.

hahaha :(

Reply:
point taken :D